src/parser.rs

```diff
@@ -10,6 +11,8 @@ use reader::{Reader, ReaderOffset};
 /// An error that occurred when parsing.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum Error {
+    /// An I/O error occurred while reading.
+    Io,
```

Comment:
It's not just `Copy` that is the problem. We would have to remove, `Clone`, `PartialEq` and `Eq` too. We currently need `Clone` or `Copy` for returning errors that are stored in `EvaluationState`; not sure if that is fixable. We use `Eq` in our tests, but that could be avoided by using `match` instead. It's a bigger change than I want to do right now though.